Connect wallet button prefab

0xFirekeeper opened the pull request:
Vanilla version of Prefab_ConnectWallet

0xFirekeeper commented:
Adding Enums.cs at the SDK level might help bridge the gap between high level calls and inner classes

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -0,0 +1,256 @@
+    private void Awake()
+    {
+#if !UNITY_EDITOR
+        SDK = new ThirdwebSDK(chain.ToString().ToLower());
```
this wont work with chains names that have a `-` like `optimism-goerli`. One thing that might help for this is that we're about to merge a PR in the TS SDK that now accepts ChainID as the network input. So that will work nicely with an enum. For now I think passing string is safer

0xFirekeeper replied:
Can you point me to where I can find the expected strings? But yes chain ID as the network input would solve it.

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -0,0 +1,256 @@
+
+    void LogThirdweb(string _message)
+    {
+        Debug.Log($"[Thirdweb] {_message}");
```
one thing that ive been thinking of is a global flag to enable / disable logs in the SDK. Wonder if these might get a bit verbose? fine for now tho

0xFirekeeper replied:
Having logging as an option is probably a good idea, would remove a bunch of logs the dev would typically add anyway.

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -0,0 +1,256 @@
+        Vector3 targetScale = dropdownExpand ? Vector3.one : Vector3.zero;
+        while (Math.Abs(_object.localScale.x - targetScale.x) > 0.01f)
+        {
+            _object.localScale = Vector3.Lerp(_object.localScale, targetScale, 10f * Time.deltaTime);
```
prob don't need to animate here. if we do then let's make it a simple fade

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -0,0 +1,256 @@
+               });
+
+            connectInfoText.text = $"{_wallet} ({chain})";
+            walletAddressText.text = $"Connected As: {address}";
```
can we shorten the address? it makes the button really large otherwise. We can have a util that shortens to `0x123...456`

0xFirekeeper commented:
Added `ShortenAddress` to Utils, simplified the rest of the code. 
Chain enum with custom values as chain ID will be great in the future once supported!

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -46,22 +26,21 @@ public struct WalletButton
-    public List<Wallet> supportedWallets;
+    public string chain = "ethereum";
+    public int chainID = 1;
+    public List<Wallet> supportedWallets = new List<Wallet> { Wallet.MetaMask, Wallet.CoinbaseWallet, Wallet.Injected, Wallet.MagicAuth };
```
remove Injected / Magic, add WalletConnect - default should be MM / CBW / WC

joaquim-verges commented on Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs:
```diff
@@ -170,66 +137,14 @@ public void OnTryConnect(string _walletStr)
 
     // Utility
 
-    int GetChainID(Chain _chain)
```
wonder if it's worth leaving this but using string (could also help validate the inputted string)

just until we ship the new chain id SDK input
